packages/tools/devtools/devtools-core/src/FluidDevtools.ts

```diff
@@ -145,6 +148,22 @@ export class FluidDevtools implements IFluidDevtools {
 			this.postContainerList();
 			return true;
 		},
+		[ToggleUnsampledTelemetry.MessageType]: async (message) => {
+			console.log("Unsampled telemetry message received", message);
+			if (message === undefined) {
+				console.error(
+					"Invalid value received in ToggleUnsampledTelemetry message:",
+					message,
+				);
+				return false;
+			}
+			const newValue = (message as ToggleUnsampledTelemetry.Message).data.unsampledTelemetry;
+			console.log("Unsampled telemetry new value:", newValue);
+			sessionStorage.setItem("Fluid.Telemetry.DisableSampling", String(newValue));
+			this.postSupportedFeatures();
+			window.location.reload();
+			return true;
+		},
```

Comment:
A couple of things here:
- Let's remove the console logs in the final version.
- Did you hit a case where the incoming message was undefined? Since we control them I wouldn't expect to be able to run into that. Seems fine to have as a defensive check, just curious if it was preemptive or in response to something that actually happened.

Reply:
Yeah I kept it as a defensive check in case it came back undefined.